spec/features/work_packages/share_spec.rb

```diff
@@ -326,6 +375,66 @@ def inherited_member_roles(group:)
       share_modal.expect_not_shared_with(new_user)
       share_modal.expect_shared_count_of(6)
     end
+
+    it 'allows creating multiple users at once' do
+      share_modal.expect_open
+      share_modal.expect_shared_count_of(6)
+
+      # Invite two users that does not exist yet
+      share_modal.create_and_invite_user(['hello@world.de', 'aloha@world.de'], 'Comment')
+
+      # New user is shown in the list of shares
+      share_modal.expect_shared_count_of(8)
+
+      # New user is created
+      new_users = User.last(2)
+
+      share_modal.expect_shared_with(new_users[0], 'Comment', position: 2)
+      share_modal.expect_shared_with(new_users[1], 'Comment', position: 1)
+
+      # The new users can be interacted with
+      share_modal.change_role(new_users[0], 'View')
+      share_modal.expect_shared_with(new_users[0], 'View', position: 2)
+      share_modal.change_role(new_users[1], 'View')
+      share_modal.expect_shared_with(new_users[1], 'View', position: 1)
+      share_modal.expect_shared_count_of(8)
+
+      # The new users can be updated simultaneously
+      share_modal.invite_user(new_users, 'Edit')
+      share_modal.expect_shared_with(new_users[0], 'Edit', position: 2)
+      share_modal.expect_shared_with(new_users[1], 'Edit', position: 1)
+      share_modal.expect_shared_count_of(8)
+
+      # The new users can be deleted
+      share_modal.remove_user(new_users[0])
+      share_modal.expect_not_shared_with(new_users[0])
+      share_modal.remove_user(new_users[1])
+      share_modal.expect_not_shared_with(new_users[1])
+      share_modal.expect_shared_count_of(6)
+    end
+
+    it 'allows sharing with an existing user and creating a new one at the same time' do
+      share_modal.expect_open
+      share_modal.expect_shared_count_of(6)
+
+      # Add an existing and a non-existing user to the autocompleter
+      share_modal.select_existing_user not_shared_yet_with_user
+      share_modal.select_not_existing_user_option "hello@world.de"
+
+      share_modal.select_invite_role('View')
+      within share_modal.modal_element do
+        click_button 'Share'
+      end
```

Comment:
What do you think about:
```ruby
# Example usages:
share_modal.invite_users([not_shared_yet_with_user, "hello@world.de"], 'View')
share_modal.invite_user(not_shared_yet_with_user, 'Edit')
share_modal.invite_user("hello@world.de", 'Edit')

# in spec/support/components/work_packages/share_modal.rb
def invite_user(users, role_name)
  Array(users).each do |user|
    case user
    when String
      select_not_existing_user_option(user)
    when Principal
      select_existing_user(user)
    end
  end

  select_invite_role(role_name)

  within modal_element do
    click_button 'Share'
  end
end

alias_method :invite_users, :invite_user
```

or if we prefer to keep the actions a bit more granular:

```ruby
# Example usages:
share_modal.autocomplete(not_shared_yet_with_user)
share_modal.autocomplete('hello@world.de')
share_modal.autocomplete([not_shared_yet_with_user, other_user, 'hello@world.de'])
### and then
share_modal.select_invite_role('View')
share_modal.click_share

# in spec/support/components/work_packages/share_modal.rb
def autocomplete(user)
  case user
  when String
    select_not_existing_user_option(user)
  when Principal
    select_existing_user(user)
  end
end

def click_share
  within modal_element do
    click_button 'Share'
  end
end
```
This could reduce the amount of methods we maintain and keep the expectations at a higher level resembling the user interactions we're performing.